lib/bundler/cli/change.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Bundler
+  class CLI::Change
+    def initialize(options, gem_name)
+      @gem_name = gem_name
+      @options = options
+    end
+
+    def run
+      raise InvalidOption, "Please supply at least one option to change." unless @options[:group] || @options[:version] || @options[:source]
+
+      definition = Bundler.definition
+
+      @dep = definition.dependencies.find {|d| d.name == @gem_name }
+
+      raise InvalidOption, "`#{@gem_name}` could not be found in the Gemfile." unless @dep
+
+      @pass_options = {}
+
+      initial_gemfile = IO.readlines(Bundler.default_gemfile)
+
+      set_group_options
+
+      set_version_options
+
+      @pass_options["source"] = @options[:source] if @options[:source]
+
+      begin
+        require "bundler/cli/remove"
+        CLI::Remove.new([@gem_name], {}).run
+
+        require "bundler/cli/add"
+        CLI::Add.new(@pass_options, [@gem_name]).run
+      rescue StandardError => e
+        SharedHelpers.write_file(Bundler.default_gemfile, initial_gemfile)
+        FriendlyErrors.log_error(e)
```

Comment:
I think instead of this, we should just `raise`

Reply:
After rescue then again we should raise error?

Reply:
yes -- the rescue here is only to restore the initial gemfile